### ISSUE TITLE
incorrect .merlin but no warning when using per_module

### ISSUE THREAD

**Issue by Khady** [ocaml/dune#2596]
We have a library using `per_module` in the preprocess part. It seems that dune generates an incorrect .merlin file as soon as `per_module` is used. It doesn't include the `FLG -ppx ...` line.

An easy way to reproduce (with dune 1.10 at least) is to create a library with a single file and this dune file:

```lisp
(library
 (name demo)
 (preprocess (per_module ((pps lwt_ppx ppx_expect) demo))))
```

**Comment by diml** [comment]
Yh, it's because the merlin ppx config is global to a directory. We have a plan with @trefis to change the way dune communicates the config to merlin and in particular make it per-file rather than per-directory.

**Comment by Khady** [comment]
I understand that there is a plan like this. I think it was mentioned in the issue which led to the introduction of the warning when dune can't generate a full .merlin file. Here we were surprised because there is no such warning, dune is silent about the problem.

**Comment by diml** [comment]
Ah indeed. That seems wrong. I suppose we wanted to fix the problem where one has several `library` stanzas in the same directory but didn't consider this case. We should emit a warning in this case as well.

**Comment by voodoos** [comment]
@Khady this should be fixed with dune 2.8.0, can you confirm it ?

**Comment by rgrinberg** [comment]
Why was this re-opened?
